pkg/memio/ports_linux.go

```diff
@@ -13,16 +13,28 @@ import (
 	"syscall"
 )
 
-const portPath = "/dev/port"
+var portPath = "/dev/port"
 
-var ioplError struct {
-	sync.Once
-	err error
-}
+var (
+	ioplError struct {
+		sync.Once
+		err error
+	}
+
+	// syscallIopl enables Mocking of syscall function
+	syscallIopl = syscall.Iopl
+	// Mocking of assembly functions
```

Comment:
the globals are scaring me. Could we consider this
type IO interface {
Outl() error
Outw() 
Outb()
inl() (uint32, error)
...
}

then to use the real thing
type ioport struct {
f*os.File
}
implement the IO interface for this with existing functions

Then for mocks
type mocker struct {
whateverneed
}
then implement IO interface for the mocker

This is a bit more idiomatic; these globals are starting to feel a bit too much like C.

Ryan, please comment, I need to be sure I'm not crazy.